### ISSUE TITLE
local vs. deployed: width without units works locally, but not when hosted

### ISSUE THREAD

**Issue by ghchinoy** [google/mesop#498]
**Describe the bug**

`me.image(style=me.Style(width="150")` is honored (width 150 px) when running mesop locally   but not when on hosted on cloud run

`style="width: 150px;"` is present on the img elements with local mesop, but not when deployed.
Changing this to `me.image(style=me.Style(width="150px")` works as expected locally and deployed.

**To Reproduce**

1. create a series of images with a unit-less width
2. run mesop locally, see that the width is honored
3. deploy to cloud run, and see that the width is not
4. view source to see that locally, the units are added, but when hosted, the units are not, which causes original image size to be displayed.

**Expected behavior**

Some consistency between local and hosted - either error in local version or assume units for the hosted version.

**Screenshots**

![image](https://github.com/google/mesop/assets/469685/9a04cc9b-0c97-4378-a566-aa3575b19686)

![image](https://github.com/google/mesop/assets/469685/52b7c08f-cfb7-432e-8ace-82ab04312c04)

**Desktop System Info**

**Additional context**

**Comment by wwwillchen** [comment]
Thanks for filing this issue.

I investigated it and it's definitely unexpected behavior. The root cause is the Pydantic dependency.

The TLDR is that if you use pydantic v2 - both locally and hosted - then you will get the strict behavior where width without units does _not_ work. My guess is that you were using Pydantic v1 in your local environment and Pydantic v2 in your hosted environment.

If you use pydantic v1 (both locally and hosted), then it follows non-strict behavior where width without units _does_ work (e.g. `width="150"`), however, I would not rely on this and I would use `width=150` or `width=150px`, which is more explicit. The reason the loose behavior works in [Pydantic v1 has non-strict behavior](https://docs.pydantic.dev/1.10/usage/validation_decorator/#coercion-and-strictness) but [Pydantic v2 has strict behavior by default](https://docs.pydantic.dev/latest/migration/#validator-behavior-changes).

I think the strict behavior is the right thing because the type coercion here is confusing (in Pydantic v1 `"150"` is implicitly turned into the int `150` which Mesop treats as pixel values).

Unfortunately, it's not straightforward to fix this in Mesop because Pydantic v2 is heavily used in the open-source world, but Pydantic v1 is used internally within Google, so we need to support both versions.

I'll leave this issue open and see if other people also hit it. If it's a frequent issue, we could add additional warnings, but it's quite a bit of effort and I suspect most people outside of Google won't hit this because they're using Pydantic v2 everywhere.
